### ISSUE TITLE
LoadMap

### ISSUE THREAD

**Issue by urban119** [otland/forgottenserver#1796]
Hello
It is possible, to loadMap as specified start position x,y,z ?

If i use loadMap(data/map.otbm) it imposes two map... 
Can You direct me, what function i need edit, when map Starts load?

Maybe class Map:
void setTile(uint16_t x, uint16_t y, uint8_t z, Tile* newTile);
		void setTile(const Position& pos, Tile* newTile) {			
			setTile(pos.x, pos.y, pos.z, newTile);
}

Regards

**Comment by marksamman** [comment]
No, it's not possible, but loadMap could be extended to allow offset arguments.

**Comment by EvilHero90** [comment]
do you just want to load a certain part of the map and re create it somewhere else, or what are you trying to achieve?

**Comment by urban119** [comment]
I try load certain part of the map in Main map. Just put a little fragment... 
It is possible, because second fragment a map been added to Main map, started as x,y (because position xyz on loaded second map has been the same in Main map when it been loaded). I just need moved example x + 50, y + 50 loading fragment map.

**Comment by ranisalt** [comment]
That idea is very nice.

**Comment by Schiffers** [comment]
enjoying at the comments about map, how I can load maps later of the load of the main map, I want to load anothers maps with spawns, I can try "copy" the function that load the main map to load anothers maps with spawns?

**Comment by EvilHero90** [comment]
actually you can do that pretty easy completly in lua aswell, I've once written what you suggested here using coroutines and such, so it didn't affect server performence and such
here's the youtube link https://youtu.be/sqSWUJy_wa4
If you want it feel free to hit me up.

**Comment by totolol123** [comment]
Nice, would be very useful

**Comment by urban119** [comment]
I search function, to add offset and moved generated fragment map. Why i have problem and want this to do? Becouse if I am load more copy of fragment map, I am need do loadmap with offset + 1000 example. If i do that, i can load example x5 fragment map for x5 team. Each of the teams have separate map for myself. (other thema is delete nousing map and other clearing function, but not yet to speak).
I do that in luascript.... but i wanna do werry fast, accurate and without errors load map... and source give exacly that for me.
Youtube for You:
https://youtu.be/xkvfVS5VZIg

**Comment by EvilHero90** [comment]
Well my function does exactly what you are asking for, it's combined with an indexing system to be able to locate all maps with their coordinates, you have to keep in mind if you copy a map and it contains levers and such you jeed to provide the scripts the offset aswell or it wont work at all

**Comment by urban119** [comment]
How to contact you @EvilHero90 ?

**Comment by EvilHero90** [comment]
Skype: evilhero19, I'm still at work tho

**Comment by urban119** [comment]
Okey I am end that work and say It is **Done**!
Watch me video 0:48:
https://youtu.be/gNEJnYDhcT8

I don't want do that functions with lua files.... I am edit only source and me new function loadMapU.
It is exacly what i want ;)

**Comment by ranisalt** [comment]
Lua performance is generally not an issue. Please do consider the effort of recompiling everytime and the extra issues of lower level programming of C++ when choosing not to use Lua.

**Comment by totolol123** [comment]
@urban119  can you do a pull request with it?

**Comment by urban119** [comment]
https://github.com/otland/forgottenserver/pull/1833

**Comment by urban119** [comment]
#1837

**Comment by Codex-NG** [comment]
I've accomplished a lot more with c++ than I ever have with lua, sometimes editing or making a slight change in the source is much easier and less time consuming than writing a complex script.

**Comment by Tofame** [comment]
@DSpeichert close this issue please, as you closed PR that would have solved it: https://github.com/otland/forgottenserver/pull/1837
There is no point in holding issues that only increase the count and are up for years.
